library/src/main/java/com/bumptech/glide/load/model/DirectResourceLoader.java

```diff
@@ -134,8 +150,45 @@ public Class<InputStream> getDataClass() {
     public void teardown() {}
   }
 
-  private static final class ResourceDataFetcher<DataT extends Closeable>
-      implements DataFetcher<DataT> {
+  /**
+   * Handles vectors, shapes and other resources that cannot be opened with
+   * Resources.openRawResource. Overlaps in functionality with {@link ResourceDrawableDecoder} and
+   * {@link com.bumptech.glide.load.resource.bitmap.ResourceBitmapDecoder} but it's more efficient
+   * for simple resource loads within a single application.
+   */
+  private static final class DrawableFactory
+      implements ModelLoaderFactory<Integer, Drawable>, ResourceOpener<Drawable> {
+
+    private final Context context;
+
+    DrawableFactory(Context context) {
+      this.context = context;
+    }
+
+    @Override
+    public Drawable open(Resources resources, int resourceId) {
+      return DrawableDecoderCompat.getDrawable(context, resourceId, resources.newTheme());
+    }
+
+    @Override
+    public void close(Drawable data) throws IOException {}
+
+    @Override
+    public Class<Drawable> getDataClass() {
+      return Drawable.class;
+    }
+
+    @NonNull
+    @Override
+    public ModelLoader<Integer, Drawable> build(@NonNull MultiModelLoaderFactory multiFactory) {
+      return new DirectResourceLoader<>(context, this);
+    }
+
+    @Override
+    public void teardown() {}
```

Comment:
**Suggestion**: Review the necessity of the empty close method in DrawableFactory. If Drawable resources do not require specific cleanup, document the rationale clearly to avoid confusion. <br> **Code Suggestion**: 
 ```
 + // No specific cleanup required for Drawable resources
 + // This method is intentionally left blank
 ```